app/scripts/modules/core/serverGroup/serverGroupReader.service.ts

```diff
@@ -0,0 +1,53 @@
+import {module} from 'angular';
+
+import {API_SERVICE, Api} from 'core/api/api.service';
+import {ServerGroup} from 'core/domain';
+
+export class ServerGroupReaderService {
+
+  static get $inject(): string[] {
+    return ['$log', 'API'];
+  }
+
+  constructor(private $log: ng.ILogService, private API: Api) {}
+
+  public getScalingActivities(serverGroup: ServerGroup): ng.IPromise<any[]> {
+    return this.API
+      .one('applications')
+      .one(serverGroup.app)
+      .all('clusters')
+      .all(serverGroup.account)
+      .all(serverGroup.cluster)
+      .one('serverGroups', serverGroup.name)
+      .all('scalingActivities')
+      .withParams({
+        region: serverGroup.region,
+        provider: serverGroup.cloudProvider
+      })
+      .getList()
+      .then((activities: any[]): any[] => {
+        return activities;
+      })
```

Comment:
Probably don't need this `then` block, since it's just returning the value. Looks like it was there before to handle the promise rejection, but that's handled here with the `catch`